mqtt/mqtt-bridge/src/upstream/rpc/local.rs

```diff
@@ -0,0 +1,203 @@
+use async_trait::async_trait;
+use bson::{doc, Document};
+use bytes::buf::BufExt;
+use lazy_static::lazy_static;
+use mqtt3::Event;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::{EventHandler, Handled},
+    pump::{PumpHandle, PumpMessage},
+    upstream::{CommandId, RemoteUpstreamPumpEvent, RpcCommand},
+};
+
+use super::RpcError;
+
+/// An RPC handlers that reposible to connect to part of the bridge which
+/// connects to local broker. It receives RPC commands on a special topic,
+/// converts it to a `RpcCommand` and sends to remote pump as a `PumpMessage`.
+pub struct LocalRpcHandler {
+    remote_pump: PumpHandle<RemoteUpstreamPumpEvent>,
+}
+
+impl LocalRpcHandler {
+    pub fn new(remote_pump: PumpHandle<RemoteUpstreamPumpEvent>) -> Self {
+        Self { remote_pump }
+    }
+}
+
+#[async_trait]
+impl EventHandler for LocalRpcHandler {
+    type Error = RpcError;
+
+    async fn handle(&mut self, event: &Event) -> Result<Handled, Self::Error> {
+        if let Event::Publication(publication) = event {
+            if let Some(command_id) = capture_command_id(&publication.topic_name) {
+                let doc = Document::from_reader(&mut publication.payload.clone().reader())?;
+                match bson::from_document(doc)? {
```

Comment:
```if let``` instead of match? Guessing there will be more though.

Reply:
iirc clippy was a bit opionated on the structure